learning_journal_basic/learning_journal_basic/tests.py

```diff
@@ -1,29 +1,219 @@
-import unittest
-
+import pytest
+import transaction
+import os
+import faker
+import random
 from pyramid import testing
+from .models import (
+    MyEntry,
+    get_engine,
+    get_session_factory,
+    get_tm_session,
+    )
+from .models.meta import Base
+from datetime import datetime
+from pyramid.httpexceptions import HTTPNotFound
+
+TITLE = ["learning journal 1", "learning journal 2", "learning journal 3"]
+
+FAKE = faker.Faker()
+
+ENTRIES = [MyEntry(
+    title=random.choice(TITLE),
+    blog_entry=FAKE.text(100),
+    creation_date=datetime.utcnow()
+    )
+        for i in range(3)]
 
+DB_SETTINGS = "sqlite:///:memory:" #setting DB (gloBAL VARIBLE)
 
-class ViewTests(unittest.TestCase):
-    def setUp(self):
-        self.config = testing.setUp()
+@pytest.fixture(scope="session")
+def setup_test_env():
+    os.environ[ #telling where the database_url test is
+        "DATABASE_URL"
+        ] = "sqlite:///:memory:"
 
-    def tearDown(self):
+@pytest.fixture(scope="function")
+def sqlengine(request):
+    config = testing.setUp(settings={
+        "sqlalchemy.url": DB_SETTINGS
+        # "sqlite:///:memory:"
+    })
+    config.include(".models")
+    config.include(".routes")
+    settings = config.get_settings()
+    engine = get_engine(settings)
+    Base.metadata.create_all(engine)
+
+    def teardown():
         testing.tearDown()
+        transaction.abort()
+        Base.metadata.drop_all(engine)
+
+    request.addfinalizer(teardown)
+    return engine
+
+@pytest.fixture(scope="function")
+def new_session(sqlengine, request):
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    def teardown():
+        transaction.abort()
+    request.addfinalizer(teardown)
+    return session
+
+def test_model_get_added(new_session):
+    assert len(new_session.query(MyEntry).all()) == 0
+
+@pytest.fixture()
+def dummy_request(new_session):
+    """creating a dummy request."""
+    d_request = testing.DummyRequest(dbsession=new_session)
+    return d_request
+
+#------------------unittest-------------------------
```

Comment:
Consider including fixtures with the test section or create sections for fixtures to keep fixtures and tests separate and identifiable.